File: gen_scripts/python3/scripts/e40.branch.py
#!/usr/bin/env python3
import sys  # system library for standard input and output

# build a menu string to output to user
menu = """\
Select an item from the menu.

  1 - Coffee
  2 - Espresso
  3 - Latte
  4 - Machiato
  5 - Capucino
  6 - Mocha
  7 - Tea

Make your selection: """

# get user output
sys.stdout.write(menu)              # output menu and prompt
selection = int(sys.stdin.read(1))  # read one number

# multiway branch using match/case (Python 3.10+) - see e30/e31.branch.py
#  for the if/elif equivalent
match selection:
  case 1:
    print("You selected a Coffee")
  case 2:
    print("You selected an Espresso")
  case 3:
    print("You selected a Latte")
  case 4:
    print("You selected a Machiato")
  case 5:
    print("You selected a Capucino")
  case 6:
    print("You selected a Mocha")
  case 7:
    print("You selected a Tea")
  case _:
    print("You have not entered a valid selection")

File: lessons/win_scripts/wsh.jscript/scripts/e40.branch.js
// JScript has no heredoc/triple-quote syntax - a "+"-continued chain of
//  single-quoted lines is the closest stand-in for a multi-line string
//  literal, all held in one variable and printed with a single call.
// The last line has no trailing "\n" - no linebreak before the answer.
var menu = 'Select an item from the menu.\n' +
           '\n' +
           '  1 - Coffee\n' +
           '  2 - Espresso\n' +
           '  3 - Latte\n' +
           '  4 - Machiato\n' +
           '  5 - Capucino\n' +
           '  6 - Mocha\n' +
           '  7 - Tea\n' +
           '\n' +
           'Make your selection: ';
WScript.stdout.write(menu);
selection = WScript.stdin.readline();

switch (selection) {
   case "1": WScript.Echo("You selected a Coffee"); break;
   case "2": WScript.Echo("You selected an Espresso"); break;
   case "3": WScript.Echo("You selected a Latte"); break;
   case "4": WScript.Echo("You selected a Machiato"); break;
   case "5": WScript.Echo("You selected a Capucino"); break;
   case "6": WScript.Echo("You selected a Mocha"); break;
   case "7": WScript.Echo("You selected a Tea"); break;
   default:  WScript.Echo("You have not entered a valid selection");
}
